### PR TITLE
Modernize documentation code examples

### PR DESCRIPTION
**Pull request by n8chur**
Updates code examples to use result builder syntax and view modifiers.

Paired with @watt and @robmaceachern for these changes.

### CODE REVIEW

File: Documentation/GettingStarted/Layout.md [hunk 1]
```diff
@@ -39,26 +59,28 @@ A `Centered` element always wraps a single child. During a layout pass, the layo
...
 ```swift
-let centered = Centered(Label(text: "Hello, world"))
+Label(text: "Hello, world")
+    .centered()
 ```
 
 ### `Aligned`
 
-Aligns a single child horizontally and vertically to the left (leading edge), right (trailing edge), top, bottom, or center of the available space. Like `Centered`, it delegates measuring to the child.
+Aligns a single child horizontally and vertically to the left (leading edge), right (trailing edge), top, bottom, or center of the available space. Like `Center`, it delegates measuring to the child.
```
Comment (robmaceachern) [inline Documentation/GettingStarted/Layout.md:68]:
```suggestion
Aligns a single child horizontally and vertically to the left (leading edge), right (trailing edge), top, bottom, or center of the available space. Like `Centered`, it delegates measuring to the child.
```

File: Documentation/Reference/ViewDescription.md [hunk 2]
```diff
@@ -55,15 +57,21 @@ let description = UIView.describe { config in
 ### Specifying a subview that should contain any view-backed child elements
 
 ```swift
-let description = MyCustomView.describe { config in
-    config.contentView = { $0.childContainerView }
+private class MyCustomView {
```
Comment (robmaceachern) [inline Documentation/Reference/ViewDescription.md:60]: Does this need to be a `UIView` subclass?

```suggestion
private class MyCustomView: UIView {
```

File: Documentation/GettingStarted/Layout.md [hunk 3]
```diff
@@ -7,25 +7,45 @@ Instead, Blueprint allows each element in the tree to use a different layout imp
...
+Label(text: "Hello, world")
+    .centered()
+    .box(background: .blue)
 ```
 
 By adding a `Centered` element into the hierarchy, the label will now be centered within the outer box.
 
+## Layout Modifiers
+
+To customize an `Element`, you call methods called modifiers. Modifiers wrap an `Element` to change its display or other properties. Each modifier returns a new `Element`, so it’s common to chain multiple modifiers, stacked vertically.
```
Comment (kyleve) [inline Documentation/GettingStarted/Layout.md:28]: Maybe

> Modifiers wrap an `Element` in another `Element`

?

File: Documentation/Tutorials/Tutorial2.md [hunk 4]
```diff
@@ -345,42 +316,35 @@ struct ReceiptElement: ProxyElement {
...
-        scrollView.alwaysBounceVertical = true
-        return scrollView
+            LineItemElement(
+                title: "Total",
+                price: purchase.total
+            )
+        }
+        .inset(uniform: 24)
+        .scrollable { scrollable in
+            scrollable.alwaysBounceVertical = true
```
Comment (robmaceachern) [inline Documentation/Tutorials/Tutorial2.md:346]:
```suggestion
        .scrollable { scrollView in
            scrollView.alwaysBounceVertical = true
```

File: Documentation/Tutorials/Tutorial2.md [hunk 5]
```diff
@@ -294,35 +274,26 @@ struct ReceiptElement: ProxyElement {
...
+            )
+
+            LineItemElement(
+                title: "Total",
+                price: purchase.total
+            )
+        }
+        .inset(uniform: 24)
+        .scrollable { scrollable in
+            scrollable.alwaysBounceVertical = true
```
Comment (robmaceachern) [inline Documentation/Tutorials/Tutorial2.md:295]:
```suggestion
        .scrollable { scrollView in
            scrollView.alwaysBounceVertical = true
```

File: Documentation/Tutorials/Tutorial2.md [hunk 6]
```diff
@@ -261,25 +250,16 @@ struct ReceiptElement: ProxyElement {
...
-                    price: purchase.total))
+        Column(alignment: .fill, minimumSpacing: 16) {
+            LineItemElement(
+                title: "Total",
+                price: purchase.total
+            )
+        }
+        .inset(uniform: 24)
+        .scrollable { scrollable in
+            scrollable.alwaysBounceVertical = true
```
Comment (robmaceachern) [inline Documentation/Tutorials/Tutorial2.md:261]:
```suggestion
        .scrollable { scrollView in
            scrollView.alwaysBounceVertical = true
```

File: Documentation/Tutorials/Tutorial2.md [hunk 7]
```diff
@@ -228,14 +218,13 @@ We'll now replace the "Hello, World" label with a `LineItemElement`:
...
-            price: purchase.total)
-
-        var scrollView = ScrollView(wrapping: totalItem)
-        scrollView.contentSize = .fittingHeight // Stretches content to fill width, but sizes height to fit
-        scrollView.alwaysBounceVertical = true
-        return scrollView
+            price: purchase.total
+        )
+        .scrollable { scrollable in
+            scrollable.alwaysBounceVertical = true
```
Comment (robmaceachern) [inline Documentation/Tutorials/Tutorial2.md:226]:
```suggestion
        .scrollable { scrollView in
            scrollView.alwaysBounceVertical = true
```

File: Documentation/Tutorials/Tutorial2.md [hunk 8]
```diff
@@ -78,18 +77,14 @@ We do this with the `Inset` element. We'll wrap the label in an `Inset` before w
...
-        scrollView.contentSize = .fittingHeight // Stretches content to fill width, but sizes height to fit
-        scrollView.alwaysBounceVertical = true
-        return scrollView
+        Label(text: "Hello, world") { label in
+            label.font = .boldSystemFont(ofSize: 24.0)
+            label.color = .darkGray
+        }
+        .inset(uniform: 24.0)
+        .scrollable { scrollable in
+            scrollable.alwaysBounceVertical = true
```
Comment (robmaceachern) [inline Documentation/Tutorials/Tutorial2.md:86]:
```suggestion
        .scrollable { scrollView in
            scrollView.alwaysBounceVertical = true
```

File: Documentation/Tutorials/Tutorial2.md [hunk 9]
```diff
@@ -51,14 +51,13 @@ Next, we'll configure the `ScrollView and return it.
...
-        var scrollView = ScrollView(wrapping: label)
-        scrollView.contentSize = .fittingHeight // Stretches content to fill width, but sizes height to fit
-        scrollView.alwaysBounceVertical = true
-        return scrollView
+        Label(text: "Hello, world") { label in
+            label.font = .boldSystemFont(ofSize: 24.0)
+            label.color = .darkGray
+        }
+        .scrollable { scrollable in
+            scrollable.alwaysBounceVertical = true
```
Comment (robmaceachern) [inline Documentation/Tutorials/Tutorial2.md:59]:
```suggestion
        .scrollable { scrollView in
            scrollView.alwaysBounceVertical = true
```

**Review by kyleve — APPROVED** [review]
Amazing! Thanks for doing this!

File: Documentation/Tutorials/Tutorial2.md [hunk 10]
```diff
@@ -531,50 +492,39 @@ struct ReceiptElement: ProxyElement {
...
+
+            LineItemElement(
+                style: .bold,
+                title: "Total",
+                price: purchase.total
+            )
+        }
+        .inset(uniform: 24)
+        .scrollable { scrollable in
+            scrollable.alwaysBounceVertical = true
```
Comment (robmaceachern) [inline Documentation/Tutorials/Tutorial2.md:526]:
```suggestion
        .scrollable { scrollView in
            scrollView.alwaysBounceVertical = true
```

**Review by robmaceachern — APPROVED** [review]
Big improvement! It looks so much nicer now.

I highlighted a few `scrollable` param names that could be updated so they are all consistent.

File: Documentation/GettingStarted/Layout.md (hunk 3 above)
Comment (n8chur) [reply Documentation/GettingStarted/Layout.md]: 035a78a

File: Documentation/GettingStarted/Layout.md (hunk 1 above)
Comment (n8chur) [reply Documentation/GettingStarted/Layout.md]: Good eye!

4af49c7

File: Documentation/Reference/ViewDescription.md (hunk 2 above)
Comment (n8chur) [reply Documentation/Reference/ViewDescription.md]: 3780d80

File: Documentation/Tutorials/Tutorial2.md (hunk 4 above)
Comment (n8chur) [reply Documentation/Tutorials/Tutorial2.md]: This was the change that didn't save during our padding!

ffb74e7

File: Documentation/Tutorials/Tutorial2.md (hunk 6 above)
Comment (n8chur) [reply Documentation/Tutorials/Tutorial2.md]: ffb74e7

File: Documentation/Tutorials/Tutorial2.md (hunk 5 above)
Comment (n8chur) [reply Documentation/Tutorials/Tutorial2.md]: ffb74e7

File: Documentation/Tutorials/Tutorial2.md (hunk 7 above)
Comment (n8chur) [reply Documentation/Tutorials/Tutorial2.md]: ffb74e7

File: Documentation/Tutorials/Tutorial2.md (hunk 9 above)
Comment (n8chur) [reply Documentation/Tutorials/Tutorial2.md]: ffb74e7

File: Documentation/Tutorials/Tutorial2.md (hunk 10 above)
Comment (n8chur) [reply Documentation/Tutorials/Tutorial2.md]: ffb74e7

File: Documentation/Tutorials/Tutorial2.md (hunk 8 above)
Comment (n8chur) [reply Documentation/Tutorials/Tutorial2.md]: ffb74e7

**Pull request merged**